File: app/src/main/java/com/elegion/android/template/data/provider/OkHttpProvider.java
package com.elegion.android.template.data.provider;

import android.support.annotation.NonNull;

import com.elegion.android.template.BuildConfig;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

/**
 * @author Mikhail Barannikov
 */
public final class OkHttpProvider {
    private OkHttpProvider() {
        throw new IllegalStateException("Final class can not be instantiated");
    }

    @NonNull
    public static OkHttpClient provideClient() {
        final OkHttpClient.Builder builder = new OkHttpClient.Builder();
        // logs
        if (BuildConfig.DEBUG) {
            builder.addInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY));
        }
        return builder.build();
    }

}

File: app/src/main/kotlin/com/elegion/android/template/data/provider/OkHttpProvider.kt
package com.elegion.android.template.data.provider

import com.elegion.android.template.BuildConfig

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

object OkHttpProvider {
    @JvmStatic
    fun provideClient(): OkHttpClient {
        val builder = OkHttpClient.Builder()
        // logs
        if (BuildConfig.DEBUG) {
            builder.addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
        }
        return builder.build()
    }
}
